### PR TITLE
Workflow Automation with Github Actions

### PR DESCRIPTION
**Pull request by alyip98**
[Preview](https://deploy-preview-187--learningresources.netlify.com/contents/projectmanagement/githubactions)

As this is a pretty rough draft, I would appreciate reviews that focus on the quality of the content and its usefulness to potential readers. 

Todo:
- [ ] ~Refactor to minimize screenshots~
- [x] Upload images to repo
- [x] Recommend useful actions such as cache
- [x] Add Table of Contents
- [x] Title Case
- [x] Github --> GitHub
- [x] How to Get Started with GitHub Actions
- [x] Point out shortcomings with the official doc in a nice way
- [ ] ~Condense benefits into list OR expand on them?~
- [ ] ~Add intro to ## How is GitHub Actions Being Used~ Section removed
- [x] Reduce repetition
- [x] Change list indexes to 1.
- [x] Remove numbers reference for runner mins

### CODE REVIEW

**Comment by damithc** [comment]
Note that GitHub Actions is very new and likely to evolve rapidly in the coming months. Try to write in a way the contents stays correct for some time e.g., minimize screenshots.

**Comment by damithc** [comment]
> I can't request reviews

Probably didn't accept the invitation to join the org last time. I've resent the invitation.

File: contents/projectManagement/githubActions.md
Comment (le0tan) [inline contents/projectManagement/githubActions.md]: Since these benefits' descriptions are relatviely short, probably consider using numbered list instead of titles?
e.g. `1. **Affordable**: ...`

File: contents/projectManagement/githubActions.md
Comment (le0tan) [inline contents/projectManagement/githubActions.md]: Consider pushing the static asset to the repo instead of linking to an external service.

File: contents/projectManagement/githubActions.md
Comment (le0tan) [inline contents/projectManagement/githubActions.md]: same as the last one, consider using static asset within repo

File: contents/projectManagement/githubActions.md
Comment (le0tan) [inline contents/projectManagement/githubActions.md]: same here

File: contents/projectManagement/githubActions.md
Comment (le0tan) [inline contents/projectManagement/githubActions.md]: Probably you may recommend some widely used actions like [cache](https://github.com/actions/cache), etc.

**Review by le0tan — CHANGES_REQUESTED** [review]

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md:13]: can add a table of contents here, refer to other PRs for examples

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md]: Change all Github -> GitHub

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md]:
```suggestion
Actions are small, reusable units in a workflow. When setting up a workflow, we can re-use actions that were already written by the community. These actions are available at the marketplace.
```

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md]: How to Get Started with GitHub Actions?

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md]: Can consider using title case for all headers

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md:43]: Perhaps you can define the `workflow`, `actions` and `triggers` here instead of at How Does GitHub Actions work? Because I think that these are the basic knowledge that the reader should know before diving into the benefits and so on.

Then the How Does GitHub Actions work section should just be Example GitHub Actions Workflow. 

My 2 cents haha

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md]: Not sure if it's a good idea to critique the official documentation...

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md]: Same here, not sure if a good idea to critique the official documentation

**Review by jamessspanggg — COMMENTED** [review]
Some initial comments:

File: contents/projectManagement/githubActions.md
Comment (alyip98) [reply contents/projectManagement/githubActions.md:13]: Just curious, is there a real need for this aside from being consistent, since the right navbar doubles as a ToC?

Comment (jamessspanggg) [reply contents/projectManagement/githubActions.md:13]: oh it will be responsive based on screen size, it will not show until ur screen is of a certain smaller size.

**Review by Tejas2805 — CHANGES_REQUESTED** [review]
We are aiming for consistency across the book. All headings need to be in titleCase. Have left comments accordingly.

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]: Easy to Use

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]:
```suggestion
### Reusable Actions
```

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]:
```suggestion
### Cross OS Support
```

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]:
```suggestion
### Docker Support
```

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]:
```suggestion
## How Does Github Actions Work?
```

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]:
```suggestion
### How is GitHub Actions Being Used?
```

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]: Can you try to shorten the heading? Too big for a heading I think.

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [reply contents/projectManagement/githubActions.md]: Whichever way you decide to go ahead, please make sure to do it in titleCase

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]:
```suggestion
### Set up a Simple Workflow (Node.js)
```

File: contents/projectManagement/githubActions.md
Comment (Tejas2805) [inline contents/projectManagement/githubActions.md]:
```suggestion
## Where to Go From Here?
```

**Comment by alyip98** [comment]
Thanks for the reviews! I have addressed all of the style/formatting issues.

@damithc I decided that I would prefer to keep the screenshots as they help break up the wall of text and effectively demonstrate just how simple it is to set up a workflow on GitHub. I do acknowledge that the feature is relatively new and is subject to changes, but GitHub has been rather stable with regards to their UI, and I believe that changes in the near future will not deviate too much from the screenshots.

@le0tan I chose to keep the heading format of the benefits to emphasize their importance within the article. 

Do let me know if you have any further suggestions/improvements

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md]: For consistency, don't use emojis and format authors/reviewers similar to other articles.

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md]: tight integration mentioned twice in the space of two sentences. Can do with just one.

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md:68]: >Integration with the GitHub ecosystem allows GitHub Actions to do more than just build/test/deploy.

Something very similar was mentioned earlier in the article. Given this is a short chapter, try to avoid repetition.

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md]: Giving specific numbers increase the risk of the article getting outdated. I'm not sure the recent GitHub change to increase the free tier makes this section outdated already, even before it is published. 
Say something like `free plan offers ample runner minutes ...`

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md:80]: These sections can be bullet points. No need to add a lot of headings.

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md:92]: Seems some explanations are missing from this section e.g., there is no intro

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md]: It's better to use `1. ` for all items. Markdown generates the correct number for each item.

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md]: This item too contains some info mentioned earlier.

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md:146]: I don't think this section adds much value, or can be replaced with just one sentence.

**Review by damithc — CHANGES_REQUESTED** [review]
Added some comments. Feels like the article is not yet 'fully baked'. Reduce repetition. Remove unnecessary headings. Fill in missing 'connecting' explanations. Make the content more compact where possible.

File: contents/projectManagement/githubActions.md
Comment (alyip98) [reply contents/projectManagement/githubActions.md:146]: This entire section? I think that readers may learn more from examples of GitHub Actions used in production compared to sandboxed examples such as the one above.

Comment (damithc) [reply contents/projectManagement/githubActions.md:146]: It's fine if they are supposed to serve as educational examples. Can keep.

**Comment by alyip98** [comment]
Thank you for the review @damithc. I apologize for the late reply, I decided to put it on hold until my finals were over. 

I've addressed most of the comments you raised. I intentionally repeated the part about GitHub integration as I believe that is its Unique Selling Point and deserves a bit of extra emphasis.

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md]: This panel title can be more aligned with the purpose. e.g., `Examples of different ways GitHub Actions can be used`

File: contents/projectManagement/githubActions.md
Comment (damithc) [inline contents/projectManagement/githubActions.md]: Avoid using `I` as more authors might join in future.

**Review by damithc — CHANGES_REQUESTED** [review]
Added a couple of comments. Let me know when you have approval from a couple of peer reviewers.

**Comment by openorclose** [comment]
Some links in the TOC for mobile don't work

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md:23]: can match this ToC with the siteNav's

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md]: *Here is a list

File: contents/projectManagement/githubActions.md
Comment (jamessspanggg) [inline contents/projectManagement/githubActions.md:109]: think can use a different type of ordered list aka (`i`, `ii`) here

**Review by jamessspanggg — COMMENTED** [review]
LGTM, just some small nits:

**Review by openorclose — APPROVED** [review]

File: contents/projectManagement/githubActions.md
Comment (alyip98) [reply contents/projectManagement/githubActions.md:109]: I prefer numbered sublists

**Review by jamessspanggg — APPROVED** [review]

**Pull request merged**
